src/epgsql.erl

```diff
@@ -298,21 +298,38 @@ sync(C) ->
 cancel(C) ->
     epgsql_sock:cancel(C).
 
-%% misc helper functions
+%% @doc Execute a function within a transaction.
 -spec with_transaction(connection(), fun((connection()) -> Reply)) ->
-                              Reply | {rollback, any()}
-                                  when
-      Reply :: any().
+                          Reply | {rollback, Reply}
+                            when Reply :: any().
 with_transaction(C, F) ->
```

Comment:
Unfortunately this change breaks backward compatibility.

What I propose is to create more generic & flexible `with_transaction(C, F, Options :: proplist() | map())`.

What should be configurable? At least

* return vs re-raise
* this 'ensure committed' routine
* maybe some options for `BEGIN` like `ISOLATION LEVEL` `READ WRITE | READ ONLY` and so on https://www.postgresql.org/docs/current/static/sql-begin.html
* maybe optional logging (see #29)

And then make `with_transaction/2` call this `with_transaction/3` with options, which allows it behave like before (for backward compatibility).